src/main/scala/zio/prelude/EitherCompose.scala

```diff
@@ -0,0 +1,32 @@
+package zio.prelude
+
+trait EitherCompose[:=>[-_, +_], :+:[+_, +_]] extends AssociativeCompose[:=>] {
+  def toLeft[A]: A :=> (A :+: Nothing)
+  def toRight[B]: B :=> (Nothing :+: B)
+  def fromEither[A, B, C](a2c: => A :=> C)(b2c: => B :=> C): (A :+: B) :=> C
+
+  def eitherCompose[A, B, C](
+    a2c: A :=> C,
+    b2c: B :=> C,
+    ab2c: (A :+: B) :=> C
+  )(implicit eqA2C: Equal[A :=> C], eqB2C: Equal[B :=> C], eqA2BC: Equal[(A :+: B) :=> C]): Boolean = {
+    val law1 = compose[A, A :+: B, C](fromEither(a2c)(b2c), toLeft) === a2c
+    val law2 = compose[B, A :+: B, C](fromEither(a2c)(b2c), toRight) === b2c
+    val law3 = fromEither(compose[A, A :+: B, C](ab2c, toLeft))(compose[B, A :+: B, C](ab2c, toRight)) === ab2c
+
+    law1 && law2 && law3
+  }
+}
+
+trait EitherComposeSyntax {
+  implicit class EitherComposeOps[A, C, :=>[-_, +_]](private val a2b: A :=> C) {
+
+    /** A symbolic alias for `fromEither`. Composes `A -> C` with `B -> C` to form `A or B -> C`. */
+    def |||[B, :+:[+_, +_]](implicit either: EitherCompose[:=>, :+:]): (=> B :=> C) => ((A :+: B) :=> C) =
+      either.fromEither(a2b)
+
+    /** Composes `A -> C` with `B -> C` to form `A or B -> C`. */
+    def fromEither[B, :+:[+_, +_]](implicit either: EitherCompose[:=>, :+:]): (=> B :=> C) => ((A :+: B) :=> C) =
+      either.fromEither(a2b)
+  }
+}
```

Comment:
For EitherCompose and BothCompose it should be possible to have instance for:
* Function1, Either, Void
* Function1, Tuple2, Unit
You would have to change Nothing to be polymorphic.

Reply:
If you combine both EitherCompose and BothCompose using the same types you would end up with MonoidalCategory and it has this nice property that it does not matter how you stack together :*: and :=> it always makes sense.

Instances mentioned ☝️ can be used as utilities to re-pack tuples and either.

Reply:
Sorry for responding so late, I've been a bit busy. How have you been doing? :smile_cat: 

I'm not sure I understand :thinking: 
There are already instances 
 * `Function[-*, +*] [type :*: = ❨_, _❩; type :--> = Function]` for `ApplicationCompose`
	* How would you make it work with `Any`? Where would you put this `Any`?
 * `Function[-*, +*] [type :+: = Either]` for `EitherCompose`
	* How would you make it work with `Nothing`? Where would you put this `Nothing`?